analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -19,20 +19,265 @@
+        PackageManager(analyzerConfig, repoConfig), CommandLineTool {
+
     class Factory : AbstractPackageManagerFactory<Bower>() {
         override val globsForDefinitionFiles = listOf("bower.json")
 
         override fun create(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfiguration) =
                 Bower(analyzerConfig, repoConfig)
     }
+
+    companion object {
+        /* We do not actually depend on any features specific to Bower version 1.8.2, but we still want to
+          stick to fixed versions to be sure to get consistent results. */
+        private const val REQUIRED_BOWER_VERSION = "1.8.2"
+        private const val SCOPE_NAME_DEPENDENCIES = "dependencies"
+        private const val SCOPE_NAME_DEV_DEPENDENCIES = "devDependencies"
+        private const val PROVIDER_NAME = "Bower"
+
+        private fun extractPackageId(node: JsonNode) = Identifier(
+                provider = PROVIDER_NAME,
+                namespace = "",
+                name = node["pkgMeta"]["name"].textValueOrEmpty(),
+                version = node["pkgMeta"]["version"].textValueOrEmpty()
+        )
+
+        private fun extractRepositoryType(node: JsonNode): String {
+            var result = ""
+
+            val repositoryNode = node["pkgMeta"]["repository"]
+            if ( repositoryNode != null ) {
+                result = repositoryNode["type"].textValueOrEmpty()
+            }
+
+            return result
+        }
+
+        private fun extractRepositoryUrl(node: JsonNode): String {
+            var result = ""
+
+            val repositoryNode = node["pkgMeta"]["repository"]
+            if (repositoryNode != null) {
+                result = repositoryNode["url"].textValueOrEmpty()
+            }
+
+            if (result.isEmpty()) {
+                result = node["pkgMeta"]["_source"].textValueOrEmpty()
+            }
+
+            return result
+        }
+
+        private fun extractRevision(node: JsonNode): String {
+            val resolutionNode = node["pkgMeta"]["_resolution"]
+            if (resolutionNode == null) {
+                return ""
+            }
+
+            var result = resolutionNode["commit"].textValueOrEmpty()
+
+            if (result.isEmpty()) {
+                result = resolutionNode["tag"].textValueOrEmpty()
+            }
+
+            return result
+        }
+
+        private fun extractVcsInfo(node: JsonNode): VcsInfo {
+            return VcsInfo(
+                    type = extractRepositoryType(node),
+                    url = extractRepositoryUrl(node),
+                    revision = extractRevision(node)
+            )
+        }
+
+        private fun extractDeclaredLicenses(node: JsonNode): SortedSet<String> {
+            val result = mutableSetOf<String>()
+
+            val license = node["pkgMeta"]["license"].textValueOrEmpty()
+            if (!license.isEmpty()) {
+                result.add(license)
+            }
+
+            return result.toSortedSet()
```

Comment:
You could use this to avoid needing `val result`:

```kotlin
return if (license.isNotEmpty()) {
    sortedSetOf(license)
} else {
    sortedSetOf<String>() // Generic type probably not required.
}
```

Reply:
Another way is:

```kotlin
return sortedSetOf<String>().apply {
    if (license.isNotEmpty()) {
        add(license)
    }
}
```

Reply:
Done.